### ISSUE TITLE
SA1013 misfires for object initializer in interpolated string

### ISSUE THREAD

**Issue by Timo-Weike** [DotNetAnalyzers/StyleCopAnalyzers#3461]
Code like
~~~ cs
var x = 1;
var y = 2;
var z = 3;
                                                v
var formatedString = $"Something {new { x, y, z }}";
~~~
triggers "SA1013 Closing brace should be followed by a space", but adding space after the `}` doesn't look right and also triggers "SA1013 Closing brace should not be preceded by a space" and the description for SA1013 states

> A closing brace should always be followed by a single space, unless it is the last character on the line, or unless it is followed by a closing parenthesis, a comma, or a semicolon.

StyleCop Version: 1.2.0-beta.406

**Comment by BDisp** [comment]
I think the correct formatting is:
```cs
var formatedString = $"Something { new { x, y, z } }";
```

**Comment by Timo-Weike** [comment]
> I think the correct formatting is:
> 
> ```cs
> var formatedString = $"Something { new { x, y, z } }";
> ```

I disagree, you would also not write
~~~ cs
Foo( new { x, y, z} );
~~~
or
~~~ cs
$"Something { x }"
~~~

Also code like
~~~ cs
$"{new object()}"
~~~
does not trigger SA1013, but
~~~ cs
                        SA1013
                 SA1009 |
                      | |
                      v v
var x = $"{new object() }";
~~~
triggers [SA1009](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1009.md) and SA1013

**Comment by BDisp** [comment]
> I disagree, you would also not write
> 
> ```cs
> Foo( new { x, y, z} );
> ```
> 

I would write:
```cs
Foo(new { x, y, z } );
```

> or
> 
> ```cs
> $"Something { x }"
> ```

if I want the space yes, otherwise;
```cs
$"Something{ x }"
```
or
```cs
$"Something{x}"
```


> Also code like
> 
> ```cs
> $"{new object()}"
> ```
> 

Inside a $"" is not mandatory, since you use the same rule on the left and on the right.

> does not trigger SA1013, but
> 
> ```cs
>                         SA1013
>                  SA1009 |
>                       | |
>                       v v
> var x = $"{new object() }";
> ```
> 
> triggers [SA1009](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1009.md) and SA1013

It must use the same rule on both side.

```cs
var x = $"{new object()}";
```
or
```cs
var x = $"{ new object() }";
```

**Comment by sharwell** [comment]
From the perspective of StyleCop Analyzers, the expected formatting here is:

```csharp
var formatedString = $"Something {new { x, y, z }}";
```
